lib/features/give/widgets/filter_suggestion_card.dart

```diff
@@ -8,70 +8,79 @@ class FilterSuggestionCard extends StatelessWidget {
   Widget build(BuildContext context) {
     final size = MediaQuery.of(context).size;
-    return ConstrainedBox(
-      constraints: const BoxConstraints.expand(
-        width: 85,
-        height: 80,
-      ),
-      child: InkWell(
-        onTap: onTap,
-        child: Card(
-          color: color,
-          child: Container(
-            padding: const EdgeInsets.only(left: 5, right: 5),
-            decoration: BoxDecoration(
-              color: isFocused ? Colors.white : color,
-              borderRadius: BorderRadius.circular(3.5),
-              border: Border(
-                top: BorderSide(
-                  color: isFocused ? color : Colors.transparent,
-                ),
-                bottom: BorderSide(
-                  color: isFocused ? color : Colors.transparent,
-                  width: 5,
-                ),
-                left: BorderSide(
-                  color: isFocused ? color : Colors.transparent,
-                ),
-                right: BorderSide(
-                  color: isFocused ? color : Colors.transparent,
+    return Visibility(
+      visible: visible,
+      child: ConstrainedBox(
+        constraints: BoxConstraints.expand(
+          width: width,
+          height: height,
+        ),
+        child: InkWell(
+          onTap: onTap,
+          child: Card(
+            color: color,
+            child: Container(
+              padding: const EdgeInsets.only(left: 5, right: 5),
+              decoration: BoxDecoration(
+                color: isFocused ? Colors.white : color,
+                borderRadius: BorderRadius.circular(3.5),
+                border: Border(
+                  top: BorderSide(
+                    color: isFocused ? color : Colors.transparent,
+                  ),
+                  bottom: BorderSide(
+                    color: isFocused ? color : Colors.transparent,
+                    width: 5,
+                  ),
+                  left: BorderSide(
+                    color: isFocused ? color : Colors.transparent,
+                  ),
+                  right: BorderSide(
+                    color: isFocused ? color : Colors.transparent,
+                  ),
                 ),
               ),
-            ),
-            child: Column(
-              mainAxisAlignment: MainAxisAlignment.center,
-              children: [
-                Container(
-                  margin: const EdgeInsets.symmetric(
-                    vertical: 5,
-                    horizontal: 5,
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: [
+                  Container(
+                    margin: const EdgeInsets.symmetric(
+                      vertical: 5,
+                      horizontal: 5,
+                    ),
+                    child: Image.asset(
+                      isFocused ? activeIcon : icon,
+                      width: size.width * (isFocused ? 0.08 : 0.06),
```

Comment:
Also this is a bit ugly, I guess we can fix that differently and more scalable

Reply:
Yep, it was existing. I gotta change it